typings/index.d.ts

```diff
@@ -159,9 +161,23 @@ declare module 'mappersmith' {
     setHeaders(xmlHttpRequest: XMLHttpRequest, headers: Headers): void
   }
 
+  export type HTTPRequestOptions = http.RequestOptions | https.RequestOptions
+
+  export interface HTTPGatewayConfiguration {
+    configure?(options: HTTPRequestOptions): HTTPRequestOptions
```

Comment:
I know use cases here that add custom properties to the `options` object, that later get used in the callbacks below (e.g. add timers for measuring performance). 

Should we support that in the typings as well?

Like adding a `[key: string]: any` to the`HTTPRequestOptions` type.